File: framework/java/src/main/java/soda/web/bootstrap/ContextManager.java
package soda.web.bootstrap;

import java.util.Collections;
import java.util.Map;
import java.util.Optional;
import java.util.WeakHashMap;

public class ContextManager {

    private final Map<String, BootstrapContext> contextMap = Collections.synchronizedMap(new WeakHashMap<>());

    public BootstrapContext register(String key, byte[] artifact) {
        var context = new BootstrapContext(key, artifact);
        contextMap.put(key, context);
        return context;
    }

    public Optional<BootstrapContext> get(String key) {
        return Optional.ofNullable(contextMap.get(key));
    }

}

File: framework/kotlin/src/main/kotlin/soda/kotlin/web/bootstrap/ContextManager.kt
package soda.kotlin.web.bootstrap

import java.util.*

class ContextManager {

    private val contextMap = Collections.synchronizedMap(WeakHashMap<String, BootstrapContext>())

    fun register(key: String, artifact: ByteArray): BootstrapContext {
        val context = BootstrapContext(key, artifact)
        contextMap[key] = context
        return context
    }

    fun get(key: String): BootstrapContext? {
        return contextMap[key]
    }

}